### PR TITLE
Move min, max to Math module

### PR DESCRIPTION
**Pull request by mppf**
* Move min/max from ChapelBase to Math
* Remove documentation for them from UtilMisc_forDocs.
* Update the test domains/marybeth/test_compare_range according to the current output

Reviewed by @lydia-duncan - thanks!

Future work - improve domains/marybeth/test_compare_range test output to have a stacktrace.

- [x] full local testing

### CODE REVIEW

File: modules/standard/Math.chpl [hunk 1]
```diff
@@ -888,6 +888,87 @@ module Math {
     return logBasePow2(val, 1);
   }
 
+  //
+  // min and max
+  //
+
+  /* Returns the maximum value of two integer arguments.
```
Comment (lydia-duncan) [inline modules/standard/Math.chpl:895]: This feels a little crowded, especially in comparison to the previous documentation.  I would be inclined to reduce it to just the generic, two argument version and the three argument version, and "no doc" the rest, moving the "Math.NAN" information on the two argument version as a white lie.

Comment (mppf) [reply modules/standard/Math.chpl]: I left the generic two-args, any args, and param versions. Is that acceptable?

**Review by lydia-duncan — APPROVED** [review]
Awesome, thanks!

**Comment by lydia-duncan** [comment]
Oh, and be sure a `make docs` builds correctly!

**Pull request merged**